### ISSUE TITLE
pagenation not working for MongoES

### ISSUE THREAD

**Issue by dreamingo88** [Impetus/kundera#888]
1. persistence.xml:

```
<persistence-unit name="MONGO_PU">
    <provider>com.impetus.kundera.KunderaPersistence</provider>
    <class>models.Category</class>
    <exclude-unlisted-classes>true</exclude-unlisted-classes>
    <properties>
        <property name="kundera.nodes" value="localhost" />
        <property name="kundera.port" value="27017" />
        <property name="kundera.keyspace" value="xplayks" />
        <property name="kundera.dialect" value="mongodb" />
        <property name="kundera.client.lookup.class" value="com.impetus.client.mongodb.MongoDBClientFactory" />
        <property name="kundera.ddl.auto.prepare" value="update" />
        <property name="kundera.show.query" value="true" />

        <property name="kundera.indexer.class" value="com.impetus.client.es.index.ESIndexer" />
    </properties>
</persistence-unit>
```
1. Category.java:

```
@Id
@Column(name="id")
public String id;

@Column(name="name")
public String name;

@Column(name="code")
public String code;

@Column(name="description")
public String description;
```
1. createCategory() using MONGO_PU:

```
for(int i = 9000; i < 9030; i++) {
    Category ctg = new Category();
    ctg.name = "just a simple demo " + i;
    ctg.code = "democode";
    ctg.description = "just a demo, using mongodb and elasticsearch";

    em.persist(ctg);
}
```
1. findCategory() using MONGO_PU (it using elasticsearch to execute the queries auto):

```
Query q = em.createQuery("select ctg.id, ctg.name, ctg.code from Category ctg where ctg.code = 'democode'", Category.class);
q.setFirstResult(0);
q.setMaxResults(20);
List<Category> ctgList = q.getResultList(); //Here get the right list.

q.setFirstResult(20);
q.setMaxResults(20);
ctgList = q.getResultList(); //**Here the list is null. It's wrong!!!** 
```

If comment the line:
`<property name="kundera.indexer.class" value="com.impetus.client.es.index.ESIndexer" />`
It works right.

So I think there's something wrong with MongoES. Please check, thank you!

**Comment by devender-yadav** [comment]
Hi @dreamingo88,


This is a bug and we are going to fix this in the upcoming release _(expected within 10 days)_.

Thanks for your input.

-Dev

**Comment by devender-yadav** [comment]
Hi @dreamingo88 

Fix has been added in the trunk for this bug and it will be available in the 3.7 release _(expected in a week)_

-Dev

**Comment by devender-yadav** [comment]
@dreamingo88 


Fixed in Kundera 3.7.

-Dev
